src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketsTelemetry.cs

```diff
@@ -10,9 +11,22 @@ internal sealed class SocketsTelemetry : EventSource
     {
         public static readonly SocketsTelemetry Log = new SocketsTelemetry();
 
+        private PollingCounter? _connectionsEstablishedCounter;
+        private PollingCounter? _bytesReceivedCounter;
+        private PollingCounter? _bytesSentCounter;
+        private PollingCounter? _datagramsReceivedCounter;
+        private PollingCounter? _datagramsSentCounter;
+
+        private long _connectionsEstablished;
+        private long _bytesReceived;
+        private long _bytesSent;
+        private long _datagramsReceived;
+        private long _datagramsSent;
+
         [Event(1, Level = EventLevel.Informational)]
         public void ConnectStart(string? address)
         {
+            Interlocked.Increment(ref _connectionsEstablished);
```

Comment:
Accepts don't count as connections established?

Reply:
As far as I understand, we are targeting only client scenarios in #37428 so it seems we should measure only outgoing connects here. Or, at least, there should be separate counters for outgoing and incoming ones.

Reply:
That doesn't make sense to me.  Sockets are not just a client-side thing, and from a consumer perspective, "connections established" is not just Connect: case in point, the docs for Accept state "Creates a new Socket for a newly created connection."

Reply:
I think it would be better to add a separate counter then. Will do it.

Reply:
However, Accept activity tracking is not part of #37428 scope, so wouldn't it be better to create a separate issue? We anyways have a limited time and need to prioritize changes.

Reply:
> We anyways have a limited time and need to prioritize changes.

That doesn't mean we should merge inconsistent / misleading / incorrect designs.

Reply:
But, if we decide that `Accept` is to have a separate counter, then it doesn't seem inconsistent. Or, do you mean we should add `Accept` activities and counter into the scope of #37428?

Reply:
I mean we should either:
a) make this counter reflect both connects and accepts, or
b) we should rename this counter to be specific to connects and add one for accepts

Both (a) and (b) have implications for this PR.

cc: @davidfowl

Reply:
btw, (b) seems better to me, as it gives more information by distinguishing between connects and accepts.

Reply:
In general, as @stephentoub points out above, sockets are not just a client-side thing. It would be very weird to ship a set of Socket counters in 5.0 that omits "server-side" information like connections accepted. Let's make sure we have a comprehensive plan here. 

@karelz

Reply:
I agree with the point that it's not only client-side thing, but believe we need to track client and server metrics separately because it's important in some scenarios. The top example is YARP where we need distinct metrics for client-to-proxy and proxy-to-destination flows.

Reply:
I think we should add different counters for connect and accept. I thought the idea was to add the right counters for both server and client scenarios in 5.0...

Reply:
Accept telemetry added.